src/common/RefCountedObj.h

```diff
@@ -72,7 +72,7 @@ struct RefCountedObject {
   }
 
 private:
-  mutable std::atomic<uint64_t> nref;
+  mutable std::atomic<int64_t> nref;
```

Comment:
> To avoid this underflowing to a large integer with buggy code.

but i don't understand how this change would help.

Reply:
if we check `nref <= 0` to decide that all refs have been freed. This change was suggested by sage in https://github.com/ceph/ceph/pull/26348#discussion_r255311167